### ISSUE TITLE
Store endpoint state transitions

### ISSUE THREAD

**Issue by tgraf** [cilium/cilium#1952]
- [ ] Replace the current per endpoint logs section with a state transition system that keep tracks of state transitions with their cause.
- [ ] Move this state transition log to a new `endpoint logs` command to declutter `endpoint get`

**Comment by raybejjani** [comment]
@tgraf, @aanm and I  discussed this in person. We have a general direction but details will depend on my implementation (they should also correct these notes if they're wrong).
Our goal is to:
- Include a health field for an endpoint under `cilium endpoint get` that contains:
```yaml
- health: {OK, Bootstrap, Pending, Warning, Failure}
- build: {OK, Bootstrap, Pending, Warning, Failure}
- policy: {OK, Bootstrap, Pending, Warning, Failure}
- connected: {yes, no}
```

With possible meanings:
```
OK: everything is OK
Pending: OK, a change is pending
Warn: Changes failed, old config still in effect
Fail: Changes failed, no config in effect, no connectivity
```
These fields are intended to provide a user with a quick representation of the endpoint's state. They are not states themselves, but should provide a definitive status. For example: `connected: no` means the endpoint is not connected. This could be because of a variety of things, but a user should be able to check this first while diagnosing a connectivity problem.

- We will interleave the current endpoint status logs and state transitions (with transition reason) into a single linear log that you can see with `cilium endpoint logs`. The API itself will export JSON but CLI will default to a grepable 1-line scheme. This log is a debugging resource and should contain enough information to understand how an endpoint arrived at a certain condition.

**Comment by tgraf** [comment]
Solved via #2060
